ext/node/polyfills/http.ts

```diff
@@ -541,6 +545,7 @@ class ClientRequest extends OutgoingMessage {
         this.onSocket(createConnection(optsWithoutSignal));
       }
     }*/
+    this.onSocket(new FakeSocket());
```

Comment:
Let's add a test that asserts that `req.socket` and `res.socket` is not `undefined` and is actually an instance of `EventEmitter`